app/grandchallenge/notifications/filters.py

```diff
@@ -79,42 +83,50 @@ class FollowFilter(FilterSet):
     class Meta:
         model = Follow
         form = FilterForm
-        fields = ("forum", "topic", "forums_for_user", "content_type")
+        fields = ("forum", "forumtopic", "forums_for_user", "content_type")
 
     def search_filter(self, queryset, name, value):
         model_name = name
         if model_name == "forum":
-            app_label = "forum"
+            app_label = "discussion_forums"
             model = Forum
-            kwargs = {"name__icontains": value}
-        elif model_name == "topic":
-            app_label = "forum_conversation"
-            model = Topic
+            kwargs = {"linked_challenge__short_name__icontains": value}
+        elif model_name == "forumtopic":
+            app_label = "discussion_forums"
+            model = ForumTopic
             kwargs = {"subject__icontains": value}
-
-        name_qs = [x.id for x in model.objects.filter(**kwargs).all()]
-
-        return queryset.filter(
-            **{"object_id__in": name_qs},
-            **{
-                "content_type__exact": ContentType.objects.filter(
-                    model=model_name, app_label=app_label
-                ).get()
-            },
-        )
+        else:
+            app_label = None
+            model = None
+            kwargs = {}
+
+        if model:
+            name_qs = [x.pk for x in model.objects.filter(**kwargs).all()]
+            extra_filter = {
+                **{"object_id__in": name_qs},
+                **{
+                    "content_type__exact": ContentType.objects.filter(

```

Comment:
Style wise there are a couple of places in this file where `.filter(...).get()` is used, `.get(...)` would suffice.

Same with this dictionary construction, why the double unpacking? Wouldn't:

```python3

extra_filter = {
  "object_id__in": name_qs,
  "content_type": ContentType.objects.get(model=model_name, app_label=app_label)
}
```

Be sufficient?

Reply:
Absolutely. No idea why I did that.